src/resolvers/Mutation/updateAdvertisement.ts

```diff
@@ -0,0 +1,125 @@
+    }
+
+    const permittedFields = [
+      "_id",
+      "name",
+      "link",
+      "type",
+      "startDate",
+      "endDate",
+    ];
+
+    // Check for unintended null values in permitted fields, if all fields are permitted
+    for (const field of Object.keys(args.input)) {
+      if (!permittedFields.includes(field)) {
+        throw new errors.InputValidationError(
+          requestContext.translate(FORBIDDEN_FIELD_UPDATE_ERROR.MESSAGE),
+          FORBIDDEN_FIELD_UPDATE_ERROR.CODE,
+          FORBIDDEN_FIELD_UPDATE_ERROR.PARAM
+        );
+      }
+      const fieldValue = (args.input as Record<string, any>)[field];
+      if (
+        fieldValue === null ||
+        (typeof fieldValue === "string" && fieldValue.trim() === "")
+      ) {
+        throw new errors.InputValidationError(
+          requestContext.translate(FIELD_NON_EMPTY_ERROR.MESSAGE),
+          FIELD_NON_EMPTY_ERROR.CODE,
+          FIELD_NON_EMPTY_ERROR.PARAM
+        );
+      }
+    }
+
+    const currentUser = await User.findOne({
+      _id: _context.userId,
+    });
+
+    if (currentUser === null) {
+      throw new errors.NotFoundError(
+        requestContext.translate(USER_NOT_FOUND_ERROR.MESSAGE),
+        USER_NOT_FOUND_ERROR.CODE,
+        USER_NOT_FOUND_ERROR.PARAM
+      );
+    }
+
+    const { startDate, endDate } = args.input;
+
+    //If startDate is less than or equal to current date
+    if (
+      startDate &&
+      new Date(startDate) <= new Date(new Date().toDateString())
+    ) {
+      throw new errors.InputValidationError(
+        requestContext.translate(START_DATE_VALIDATION_ERROR.MESSAGE),
+        START_DATE_VALIDATION_ERROR.CODE,
+        START_DATE_VALIDATION_ERROR.PARAM
+      );
+    }
+
+    //If endDate is less than or equal to startDate
+    if (startDate && endDate && new Date(endDate) <= new Date(startDate)) {
+      throw new errors.InputValidationError(
+        requestContext.translate(END_DATE_VALIDATION_ERROR.MESSAGE),
+        END_DATE_VALIDATION_ERROR.CODE,
+        END_DATE_VALIDATION_ERROR.PARAM
+      );
+    }
+
+    const updatedAdvertisement = await Advertisement.findOneAndUpdate(
+      {
+        _id: args.input._id,
+      },
+      {
+        ...(args.input as any),
+      },
+      {
+        new: true,
+      }
+    ).lean();
+
+    if (!updatedAdvertisement) {
+      throw new errors.NotFoundError(
+        requestContext.translate(ADVERTISEMENT_NOT_FOUND_ERROR.MESSAGE),
+        ADVERTISEMENT_NOT_FOUND_ERROR.CODE,
+        ADVERTISEMENT_NOT_FOUND_ERROR.PARAM
+      );
+    }
+
+    const updatedAdvertisementPayload = {
+      _id: updatedAdvertisement._id.toString(), // Ensure _id is converted to String as per GraphQL schema
+      name: updatedAdvertisement.name,
+      orgId: updatedAdvertisement.orgId.toString(),
+      link: updatedAdvertisement.link,
+      type: updatedAdvertisement.type,
+      startDate: updatedAdvertisement.startDate,
+      endDate: updatedAdvertisement.endDate,
+    };
```

Comment:
`.lean()` gives back a simple javascript object instead of mongoose document, no need for `.toString()` conversions, u can simply return the advertisement

Reply:
Whenever I do that, it gives this error 
Type '{ advertisement: InterfaceAdvertisement; }' is not assignable to type 'UpdateAdvertisementPayload'.
The types of 'advertisement._id' are incompatible between these types.
Type 'ObjectId' is not assignable to type 'string'.